ffi/dotnet-idsdk/IoticsIdLib/src/Identity.cs

```diff
@@ -0,0 +1,244 @@
+
+A simple usage of the API is in the file Main.cs
+
+*/
+public class Identity(string resolverAddress)
+{
+    public string ResolverAddress
+    {
+        get
+        {
+            return Tools.IsUrl(resolverAddress) ? Tools.RemoveTrailingSlash(resolverAddress) : resolverAddress;
+        }
+    }
+
+
+    /// <summary>
+    /// Creates a Seed that is then used to initialise the random number generator used to create private keys underpinning the identities.
+    /// Important: if a seed is forgotten or lost, the keys and ultimately, identities created with this shim will be lost forever.
+    /// </summary>
+    /// <returns>the seed string</returns>
+    public static string CreateDefaultSeed()
+    {
+        return Tools.InvokeGoFunction(() => IdLib.CreateDefaultSeed());
+    }
+
+    /// <summary>
+    /// Seeds can be retrieved by supplying the mnemonics to this method
+    /// </summary>
+    /// <param name="mnemonics">the mnemonics created with SeedBip39ToMnemonic</param>
+    /// <returns>a string with mnemonics separated by a space</returns>
+    public static string MnemonicBip39ToSeed(string mnemonics)
+    {
+        return Tools.InvokeGoFunction(() => IdLib.MnemonicBip39ToSeed(mnemonics));
+    }
+
+    /// <summary>
+    /// Seeds don't have to be remembered and they can be mapped to human readeable mnemonics easier to pin down or remembered.
+    /// </summary>
+    /// <param name="seed">a seed</param>
+    /// <returns>the mnemonics string separated by space</returns>
+    public static string SeedBip39ToMnemonic(string seed)
+    {
+        return Tools.InvokeGoFunction(() => IdLib.SeedBip39ToMnemonic(seed));
+    }
+
+    /// <summary>
+    /// Creates an agent identity. This is the identity that your agent application uses to interact with space. The DID of the agent is important to
+    /// configure in space for access control
+    /// Note that the agent takes a seed. For security purposes, it should be different than that of the user and possibly twins. 
+    /// An agent identity can't simply connect to a space. It must have an auth delegation from an user identity. Also, in order to control a twin, it must have control 
+    /// delegation on that twin.  
+    /// 
+    /// The call is idempotent. If called multiple times with the same arguments, it won't override the one present in the resolver.
+    /// </summary>
+    /// <param name="seed">a seed specific for this agent as created by CreateDefaultSeed</param>
+    /// <param name="cKeyName">a key name</param>
+    /// <param name="cId">a unique ID in the identity document referring the public key matching this identity</param>
+    /// <returns>a new agent identity.</returns>
+    public AgentIdentity CreateAgentIdentity(string seed, string cKeyName, string cId)
+    {
+        string did = Tools.InvokeGoFunction(() => IdLib.CreateAgentIdentity(ResolverAddress, cKeyName, Tools.EnsureHashPrefix(cId), seed));
```

Comment:
The `Tools.Ensure*` usage: Surely the golang layer should be performing validation like this and if not we should add it and keep the wrappers as thin as possible.

Reply:
also - validation and input sanitisation should be done as early as possible